src/library/scala/collection/immutable/ArraySeq.scala

```diff
@@ -135,6 +135,34 @@ sealed abstract class ArraySeq[+A]
     else
       ArraySeq.unsafeWrapArray(new ArrayOps(unsafeArray).slice(from, until)).asInstanceOf[ArraySeq[A]]
 
+  override def foldLeft[B](z: B)(f: (B, A) => B): B = {
+    // For ArraySeqs with sizes of [100, 1000, 10000] this is [1.3, 1.8, 1.8]x as fast
+    // as the same while-loop over this instead of unsafeArray.
+    val array = unsafeArray
```

Comment:
can we benchmark if unwrapping `unsafeArray` is actually faster than calling `apply` and `length` on the instance? we don't know the type of `unsafeArray` (it could be `int[]` or `Object[]`), so calling `.length` and `.apply` translate into calls to `j.l.reflect.Array.getLength` and `ScalaRunTime.array_apply` I believe. On the other hand, the implementing classes of `ArraySeq` (e.g. `ofInt`) know their array's type, so can access elements and length directly. on the other hand, there is a layer of indirection through a field if we call the `apply` method.

Reply:
Hi @NthPortal , yes it already has been benchmarked :) It's those benchmark results that the comment is referring to. The benchmarking code is here https://github.com/japgolly/misc/tree/scalaBM and the results are reproducible. Of course any improvements to the benchmarking code is more than welcome. There's some more discussion pending in #8929

Reply:
my mistake. thanks for being so thorough!

Reply:
No worries and thanks for the extra pair of eyes!